pkg/fs/http.go

```diff
@@ -127,7 +128,7 @@ func (hfs *webdavFS) Rename(ctx context.Context, oldName, newName string) error
 }
 
 func (hfs *webdavFS) Stat(ctx context.Context, name string) (os.FileInfo, error) {
-	fi, err := hfs.fs.Stat(hfs.ctx, name)
+	fi, err := hfs.fs.Stat(hfs.ctx, utils.EncodeUserInput(name))
```

Comment:
Is it necessary? If yes, should we also apply it on other methods?

Reply:
@zhijian-pro no alert for this line

Reply:
<img width="1259" alt="image" src="https://user-images.githubusercontent.com/31313340/157793324-ce27a792-f1c3-424e-bce9-e56851cd0f9f.png">
Here is the source of several other alerts.